docker/models/containers.py

```diff
@@ -709,7 +716,14 @@ def run(self, image, command=None, stdout=True, stderr=False,
         if exit_status != 0:
             stdout = False
             stderr = True
-        out = container.logs(stdout=stdout, stderr=stderr)
+
+        logging_driver = container.attrs['HostConfig']['LogConfig']['Type']
+
+        if logging_driver == 'json-file' or logging_driver == 'journald':
```

Comment:
I agree that hardcoding logging drivers is not the best option, however Docker API doesn't provide information about logging driver abilities. Do you think catching `docker.errors.APIError` is better approach?

```python
try:
    out = container.logs(stdout=stdout, stderr=stderr)
except docker.errors.APIError:
    out = None
```

For me the only disadvantage is that we can distinguish different types of api errors only by error message.